Question2_RecursiveSearch.java

```diff
@@ -0,0 +1,47 @@
+import java.util.Arrays;
+/**
+ * This class is designed to Search an element in an array using recursive search approaches
+ * @author Gaurav Tak
+ *
+ */
+public class Question2_RecursiveSearch {
+	/**
+	 * This method is a linear search implementation using recursion
+	 * @param input
+	 * @param currentIndex
+	 * @param elementToSearch
+	 * @return index where the search element is found
+	 */
+	public int linearSearch(int input[],int currentIndex,int elementToSearch) {
+		if (currentIndex<input.length) {											//check for current index overflow
+			if(input[currentIndex]==elementToSearch) {
+				return currentIndex;												//base case
+			} else {
+				return (linearSearch(input,currentIndex+1,elementToSearch));		//recursive call
+			}
+		} else {
+			return -1;																//returns -1 if not found or array is empty		
+		}
+	}
+	
+	/**
+	 * This method is a binary search implementation using recursion
+	 * @param input
+	 * @param startIndex gives the index of the subproblem's first element in the actual array 
+	 * @param elementToSearch
+	 * @return index where the element to be searched is found
+	 */
+	public int binarySearch(int input[],int startIndex,int elementToSearch) {
```

Comment:
For binary we assume that array is sorted but it is not assumed in question so you must explicitly sort it